tnoodle-ui/src/components/EventsTable.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import CubingIcon from "../CubingIcon";
+
+class EventsTable extends Component {
+  render() {
+    const { events } = this.props;
+    let maxRounds = Math.max.apply(
+      Math,
+      events.map(event => event.rounds.length)
+    );
+    return (
+      <div className="text-center">
+        <table className="table table-striped table-bordered table-hover table-condensed">
+          <thead className="thead-dark">
+            <tr>
+              <th scope="col" />
+              <th scope="col">Event</th>
+              {Array.from(Array(maxRounds), (e, i) => {
```

Comment:
I'd never heard of `Array.from` until now, but from some googling (https://stackoverflow.com/questions/26052699/array-from-vs-array-prototype-map), it sounds like it's useful for dealing with "array like" objects, and is maybe a bit let performant than `Array.map`. In this situation, we're dealing with a plain array (not an array-like object), so I think it would be less surprising to use `map` here instead. I'd suggest rewriting this as:

```javascript
Array(maxRounds).map((e, i) => {....
```

Reply:
Changed to `[...Array(maxRounds)].map((e, i) =>`